cmd/hub/lifecycle/deploy.go

```diff
@@ -882,14 +882,23 @@ func parametersInEnv(component *manifest.ComponentRef, componentParameters param
 		componentDir = component.Source.Dir
 	} else {
 		componentDir = filepath.Join(baseDir, component.Source.Dir)
+		if !filepath.IsAbs(componentDir) {
+			t, err := filepath.Abs(componentDir)
+			if err != nil {
+				util.Warn("Unable to set absolute path for HUB_COMPONENT_DIR (%s): %v", component.Name, err)
+				util.Warn("Falling back to %s directory: %s", component.Name, componentDir)
+			} else {
+				componentDir = t
+			}
+		}
 	}
-	if !filepath.IsAbs(componentDir) {
+
+	if !filepath.IsAbs(baseDir) {
 		t, err := filepath.Abs(componentDir)
```

Comment:
@akranga Should it be `filepath.Abs(baseDir)` instead of `filepath.Abs(componentDir)`? Currently, HUB_BASE_DIR gets the same value as HUB_COMPONENT_DIR

Reply:
@akranga fix the problem in the next commit

Reply:
Yeah, this is noted. There is one more thing appeared after careful review. We cannot resolve `componentDir` based on `baseDir` and then resolve a `baseDir`. Latest resolves exactly this